rtl/eclrtl/rtlrecord.cpp

```diff
@@ -276,11 +276,13 @@ void RtlRecord::calcRowOffsets(size_t * variableOffsets, const void * _row, unsi
 {
     const byte * row = static_cast<const byte *>(_row);
     unsigned maxVarField = (numFieldsUsed>=numFields) ? numVarFields : whichVariableOffset[numFieldsUsed];
+    size32_t varoffset = 0;
     for (unsigned i = 0; i < maxVarField; i++)
     {
         unsigned fieldIndex = variableFieldIds[i];
-        size_t offset = getOffset(variableOffsets, fieldIndex);
-        size_t fieldSize = queryType(fieldIndex)->size(row + offset, row);
+        size32_t offset = fixedOffsets[fieldIndex] + varoffset;
+        size32_t fieldSize = queryType(fieldIndex)->size(row + offset, row);
+        varoffset += fieldSize;
```

Comment:
I think this should be:

    varoffset = offset + fieldsize

and in the code below.

Reply:
Are you sure? That would result in fixedOffsets being added multiple times I think

Reply:
yes, otherwise varoffset is a running total of the variable size fields - which isn't what is required, it needs to be the offset after the nth variable size field.